### PR TITLE
Add stop callback on shutdown

### PR DESCRIPTION
**Pull request by qinghuangchan**
In this PR, I propose to add some additional handling when exiting. 

Background for needing this feature:

I've opted to store my data in pgsql. The current pgsql insertion implementation doesn't do any caching, so i've monkey patched my own caching methods in.

However, when restarting the collector, the data in cache is lost, since there is no opportunity to do clean up. 

I've added a `stop` method to the `Callback` class, as well as created a new `BackendCallback` base class which all other backend callbacks inherit. 

The purpose of adding such handling is to allow the many types of callbacks to do some clean up where necessary. 

In the `run` method of `FeedHandler`, I've added a `finally` clause to the try-except statement. This is where the code will try to call `stop` for all its feeds. 

Im not too sure if you'd like my implementation - please let me know if there's something I can change to make it better. 

- [x] - Tested
- [x] - Changelog updated
- [ ] - Contributors file updated (optional)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## Changelog
+  * Feature: Add stop callback when exiting loop, add stop method placeholder for base callbacks
```
Comment (bmoscon) [inline CHANGES.md:2]: move this under the current dev version, 1.6.2

**Review by bmoscon — CHANGES_REQUESTED** [review]
rather than put in an un-used stub, why not just check if stop is a defined method on `callback` before calling it? This would remove the need for all the added in inheritance

**Comment by qinghuangchan** [comment]
That's looks nicer indeed. changes committed.

File: CHANGES.md (hunk 1 above)
Comment (qinghuangchan) [reply CHANGES.md]: done.

**Review by bmoscon — APPROVED** [review]

**Pull request merged**